lib/production_e2e_filter.rb

```diff
@@ -0,0 +1,43 @@
+module ProductionE2EFilter
+
+  BEGIN {
+    logger = Logger.new(STDOUT)
+    begin
+      E2E_FILTER_CONF = YAML::load(IO.read(File.expand_path('./conf/e2e_filter_conf.yml', __FILE__))).with_indifferent_access
+      @emails = E2E_FILTER_CONF['prefixes']['emails']
+      @domains = E2E_FILTER_CONF['prefixes']['domains']
+      logger.info("Successfully initialized red_cross production e2e filter values")
+    rescue => e
+      logger.error("Unable to initialize red_cross production e2e filter due to #{e.message}")
+    end
+  }
+
+  def is_e2e_event?(properties)
+    begin
+      filter_email?(properties[:email]) || filter_domain?(properties[:website], properties[:storeDomain]) ? true : false
+    rescue => e
+      logger.error("Failed to check for production e2e filter values due to #{e.message}")
+      false
+    end
+  end
+
+  private
+
+  def filter_email?(email)
+    @emails.each do |test_prefix|
+      if email&.start_with?(test_prefix)
+        return true
+      end
+    end
+    false
+  end
+
+  def filter_domain?(website, store_domain)
```

Comment:
not sure its very interesting to know the actual fields names, website/store_domain/bobo_the_clown.- just send a list of parms and if anyone of them is starting with the prefix then true.
maybe use the 3 dots notation:
def call(...)

Reply:
changed it to use *args, let me know what you think